src/sage/schemes/elliptic_curves/homset.py

```diff
@@ -0,0 +1,234 @@
+it even forms a (not always commutative) ring under composition.
+
+This module encapsulates the set of homomorphisms between two given elliptic
+curves as a Sage object.
+
+.. NOTE::
+
+    Currently only little nontrivial functionality is available, but this will
+    hopefully change in the future.
+
+EXAMPLES:
+
+The only useful thing this class does at the moment is coercing integers into
+the endomorphism ring as scalar multiplications::
+
+    sage: E = EllipticCurve([1,2,3,4,5])
+    sage: f = End(E)(7); f
+    Scalar-multiplication endomorphism [7] of Elliptic Curve defined by y^2 + x*y + 3*y = x^3 + 2*x^2 + 4*x + 5 over Rational Field
+    sage: f == E.scalar_multiplication(7)
+    True
+
+::
+
+    sage: E = EllipticCurve(GF(431^2), [0,1])
+    sage: E.automorphisms()[0] == 1
+    True
+    sage: E.automorphisms()[1] == -1
+    True
+    sage: omega = E.automorphisms()[2]
+    sage: omega == 1
+    False
+    sage: omega^3 == 1
+    True
+    sage: (1 + omega + omega^2) == 0
+    True
+    sage: (2*omega + 1)^2 == -3
+    True
+
+AUTHORS:
+
+- Lorenz Panny (2023)
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2023 Lorenz Panny
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.rings.integer_ring import ZZ
+from sage.categories.morphism import Morphism
+from sage.schemes.generic.homset import SchemeHomset_generic
+from sage.schemes.elliptic_curves.ell_generic import EllipticCurve_generic
+
+
+class EllipticCurveHomset(SchemeHomset_generic):
+    r"""
+    This class represents the set of all homomorphisms between two fixed
+    elliptic curves.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(419^2), [1,0])
+        sage: E.frobenius_isogeny() in End(E)
+        True
+        sage: phi = E.isogenies_prime_degree(7)[0]
+        sage: phi in End(E)
+        False
+        sage: phi in Hom(E, phi.codomain())
+        True
+
+    Note that domain and codomain are *not* taken up to isomorphism::
+
+        sage: iso = E.isomorphism_to(EllipticCurve(GF(419^2), [2,0]))
+        sage: iso in End(E)
+        False
+    """
+    def __init__(self, E1, E2, category=None):
+        r"""
+        Construct the homset for a given pair of elliptic curves
+        defined over the same base ring.
+
+        TESTS::
+
+            sage: E = EllipticCurve('37a1')                                             # needs sage.schemes
+            sage: Hom(E, E).__class__                                                   # needs sage.schemes
+            <class 'sage.schemes.elliptic_curves.homset.EllipticCurveHomset_with_category'>
```

Comment:
```suggestion
            sage: H = Hom(E, E)
            sage: TestSuite(H).run()
```
I don't find setting the class a very useful test personally. I would rather see a test like this (although I think you need to have `skip="_test_category"` by the nature of how homsets work) and similarly for `Hom(E1, E2)`.